source/NVDAObjects/behaviors.py

```diff
@@ -366,6 +367,98 @@ def event_loseFocus(self):
 		super(Terminal, self).event_loseFocus()
 		self.stopMonitoring()
 
+
+class TerminalWithKeyboardSupport(Terminal):

```

Comment:
As mentioned in https://github.com/nvaccess/nvda/pull/9915#discussion_r308730460 rename this to `KeyboardHandlerBasedTypedCharSupport`